llama_stack/providers/inline/agents/meta_reference/openai_responses.py

```diff
@@ -169,34 +171,30 @@ class OpenAIResponsePreviousResponseWithInputItems(BaseModel):
 class OpenAIResponsesImpl:
     def __init__(
         self,
-        persistence_store: KVStore,
         inference_api: Inference,
         tool_groups_api: ToolGroups,
         tool_runtime_api: ToolRuntime,
+        responses_store: ResponsesStore,
     ):
-        self.persistence_store = persistence_store
         self.inference_api = inference_api
         self.tool_groups_api = tool_groups_api
         self.tool_runtime_api = tool_runtime_api
-
-    async def _get_previous_response_with_input(self, id: str) -> OpenAIResponsePreviousResponseWithInputItems:
-        key = f"{OPENAI_RESPONSES_PREFIX}{id}"
-        response_json = await self.persistence_store.get(key=key)
-        if response_json is None:
-            raise ValueError(f"OpenAI response with id '{id}' not found")
-        return OpenAIResponsePreviousResponseWithInputItems.model_validate_json(response_json)
+        self.responses_store = responses_store
```

Comment:
Could we change the persistence store to be a sqlstore instead? The same path could continue to work and maybe we could read back the older data if we wanted to keep backwards compatibility? But we may even kill backwards compatibility here -- I don't think we have folks relying on lots of stored state across releases yet.